source/model-generator/AnnotationFeatureTemplate.cpp

```diff
@@ -0,0 +1,130 @@
+    kind = Kind::Class;
+  } else if ("Method" == target) {
+    kind = Kind::Method;
+  } else {
+    kind = Kind::Parameter;
+    try {
+      argument = RootTemplate::from_json(value);
+    } catch (...) {
+      throw_with_nested(JsonValidationError(
+          value,
+          /* field */ std::nullopt,
+          fmt::format("valid annotation target (`Class`, `Method`, or "
+                      "`Argument(...)`), got `{}`",
+                      target)));
+    }
+  }
+  return AnnotationTarget{kind, std::move(argument)};
+}
+
+AnnotationTarget::Kind AnnotationTarget::kind() const { return kind_; }
+
+const RootTemplate& AnnotationTarget::argument() const { return *argument_; }
+
+AnnotationFeatureTemplate::AnnotationFeatureTemplate(
+    AnnotationTarget target, const DexType* dex_type,
+    const DexString* MT_NULLABLE tag,
+    const DexString* MT_NULLABLE annotation_parameter_name)
+    : target_(std::move(target)),
+      dex_type_(dex_type),
+      tag_(tag),
+      annotation_parameter_name_(annotation_parameter_name) {}
+
+AnnotationFeatureTemplate AnnotationFeatureTemplate::from_json(
+    const Json::Value& value) {
+  JsonValidation::check_unexpected_members(
+      value, {"target", "type", "tag", "annotation_parameter"});
+  AnnotationTarget port = AnnotationTarget::from_json(value["target"]);
+  const DexType* dex_type = JsonValidation::dex_type(value, "type");
+  Json::Value optional_value = value["tag"];
+  const DexString* tag = nullptr;
+  if (!optional_value.isNull()) {
+    tag = DexString::make_string(JsonValidation::string(optional_value));
+  }
+  optional_value = value["annotation_parameter"];
+  const DexString* annotation_parameter_name = nullptr;
+  if (!optional_value.isNull()) {
+    annotation_parameter_name = DexString::make_string(JsonValidation::string(optional_value));
+  }
+
+  return AnnotationFeatureTemplate{std::move(port), dex_type, tag, annotation_parameter_name};
+}
+
+const Feature* MT_NULLABLE AnnotationFeatureTemplate::instantiate(
+    const Method* method, Context& context,
+    const TemplateVariableMapping& parameter_positions) const {
+  const DexAnnotationSet* anno_set;
+  switch (target_.kind()) {
+    case AnnotationTarget::Kind::Class: {
+      const DexClass* cls = type_class(method->get_class());
+      if (!cls) {
+        return nullptr;
+      }
+      anno_set = cls->get_anno_set();
+      break;
+    }
+    case AnnotationTarget::Kind::Method:
+      anno_set = method->dex_method()->get_anno_set();
+      break;
+    case AnnotationTarget::Kind::Parameter:
+      const Root root = target_.argument().instantiate(parameter_positions);
+      anno_set = method->get_parameter_annotations(root.parameter_position());
+      if (!anno_set) {
+        return nullptr;
+      }
+      break;
+  }
+
+  const std::vector<std::unique_ptr<DexAnnotation>>& annotations =
+      anno_set->get_annotations();
+  auto annotation =
+      find_if(annotations.begin(), annotations.end(),
```

Comment:
nit: Here and below
`      std::find_if(annotations.begin(), annotations.end(),`